test/test_serial_gridreduce.cpp

```diff
@@ -75,21 +76,23 @@ TEST_F(SerialGridReductionTest, CodegenNodes) {
         //   [ iS{W}, rS{H} ]
         // then schedule as
         //   [ iBIDx{blocks_x}, iBIDy{blocks_y}, iS{A}, iS{B}, iTIDy{num_warps},
-        //   iTIDx{32}, rBIDz{blocks_z} ]
+        //   iTIDx{32}, iV{vec_size} rBIDz{blocks_z} ]
         auto tv2 = tv0->cacheAfter();
         auto tv3 = tv1->cacheBefore();
 
-        tv3->reorder({{1, 0}, {0, 1}}); // blocks_x*blocks_y*A*B*num_warps*32, H
-        tv3->split(0, 32); // blocks_x*blocks_y*A*B*num_warps, 32, H
-        tv3->split(0, num_warps); // blocks_x*blocks_y*A*B, num_warps, 32, H
-        tv3->split(0, B); // blocks_x*blocks_y*A, B, num_warps, 32, H
-        tv3->split(0, A); // blocks_x*blocks_y, A, B, num_warps, 32, H
-        tv3->split(0, blocks_y); // blocks_x, blocks_y, A, B, num_warps, 32, H
+        tv3->reorder({{1, 0}, {0, 1}}); // blocks_x*blocks_y*A*B*num_warps*32*vec_size, H
+        tv3->split(0, vec_size); // blocks_x*blocks_y*A*B*num_warps, 32, vec_size, H
+        tv3->split(0, 32); // blocks_x*blocks_y*A*B*num_warps, 32, vec_size, H
+        tv3->split(0, num_warps); // blocks_x*blocks_y*A*B, num_warps, 32, vec_size, H
+        tv3->split(0, B); // blocks_x*blocks_y*A, B, num_warps, 32, vec_size, H
+        tv3->split(0, A); // blocks_x*blocks_y, A, B, num_warps, 32, vec_size, H
+        tv3->split(0, blocks_y); // blocks_x, blocks_y, A, B, num_warps, 32, vec_size, H
         tv3->axis(0)->parallelize(ParallelType::BIDx);
         tv3->axis(1)->parallelize(ParallelType::BIDy);
         tv3->axis(4)->parallelize(ParallelType::TIDy);
         tv3->axis(5)->parallelize(ParallelType::TIDx);
-        tv3->axis(6)->parallelize(ParallelType::BIDz);
+        tv3->axis(6)->parallelize(ParallelType::Vectorize);
```

Comment:
`ParallelType::Group` might be a better choice here to signal vectorized serial reduction. I'll experiment with that. Originally I worried that lowering would detect it and convert the node to a `GroupedGridReduction` but  I think that might be the proper way to handle it actually.

Reply:
I am actually going to keep this as `Vectorize` since there is another good use for `Group` in this context. When we use split-K in a **bmm**, the batch dimension is no longer parallelized as it would be without split-K. In this case we have an outer loop in the batch dim that includes a separate main loop for each row in the batch dimension. Our current approach of syncing around _all_ non-trivial loops for serial reduction is insufficient in that case since it means we are actually serializing the entire kernel instead of just each output stage. We could explicitly mark the inner dimensions as `ParallelType::Group` in order to signal that we actually want to sync inside that outer loop.